### ISSUE TITLE
Shellcheck unable to check trap arguments with double quotations

### ISSUE THREAD

**Issue by arafatx** [koalaman/shellcheck#2131]
#### For bugs
- Rule Id: SC2064
- [✓] The rule's wiki page already cover this: https://shellcheck.net/wiki/SC2064
- [✓ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [✓] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
_trap_with_args() {
  local func
  func="$1"
  shift
  for sig; do
  trap "${func} ${sig}" "${sig}" # <-- problem is here
  done
}

clean() {
  exit
}

_trap_with_args clean QUIT INT TERM EXIT

```

#### Here's what shellcheck currently says:

```
Use single quotes, otherwise this expands now rather than when signalled.
See SC2064.
```

#### Here's what I wanted or expected to see:

Nothing because if  I follow what shellcheck is currently suggesting by changing this line:

```
 trap "${func} ${sig}" "${sig}"
```
into this

```
 trap '${func} ${sig}' '${sig}'
```

this will suppress the shellcheck warning but the trap command will not work and produces the following error:

```
./test2: line 7: trap: ${sig}: invalid signal specification
./test2: line 7: trap: ${sig}: invalid signal specification
./test2: line 7: trap: ${sig}: invalid signal specification
./test2: line 7: trap: ${sig}: invalid signal specification

```

**Comment by arafatx** [comment]
I understand that I can suppress this with shellcheck Directive but, I'm trying to avoid having shellcheck comment in the code for readability especially if I want to send the code to someone who doesn't have shellcheck.

**Comment by arafatx** [comment]
I understand that I can suppress this with shellcheck Directive but, I'm trying to avoid having shellcheck comment for ignoring rules in the code for readability especially if I want to send the code to someone who doesn't have shellcheck. They might be asking, what the heck does this line do.

**Comment by matthewpersico** [comment]
Then explain it to them: 

```
## 'shellcheck' is a bash linter. 'shellcheck' reads the comments starting with 
## the text 'shellcheck' to control the linter.
## The comments have no effect on running your code.
```

**Comment by arafatx** [comment]
@matthewpersico I guess using Directive to suppress this error is the only solution. thanks.
